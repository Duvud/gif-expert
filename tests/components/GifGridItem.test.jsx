import { render, screen } from "@testing-library/react";
import GifGridItem from "../../src/components/GifGridItem";

describe('Tests sobre <GifGridItem />', () => { 

    const defaultTestTitle = 'default title';
    const defaultTestUrl = 'http://localhost/default%20url';


    test('Debe de coincidir con el snapshot', () => { 
        const {container} = render(
            <GifGridItem 
                title={defaultTestTitle}
                url={defaultTestUrl}
            />
        );
        expect(container).toMatchSnapshot();
     });

     test('Debe de mostrar la imagen con el URL y el ALT indicado', () => { 
        render(
            <GifGridItem 
                title={defaultTestTitle}
                url={defaultTestUrl}
            />
        );
        const testImg = screen.getByRole('img');
        expect(testImg.src).toBe(defaultTestUrl);
        expect(testImg.alt).toBe(defaultTestTitle);
      });

      test('Debe de mostrar el tÃ­tulo del componente', () => {
        render(
            <GifGridItem 
                title={defaultTestTitle}
                url={defaultTestUrl}
            />
        );
        expect(screen.getByText(defaultTestTitle)).toBeTruthy();
      })
 });