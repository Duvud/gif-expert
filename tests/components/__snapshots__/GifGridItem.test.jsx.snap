// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests sobre <GifGridItem /> Debe de coincidir con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="default title"
      src="http://localhost/default%20url"
    />
    <p>
      default title
    </p>
  </div>
</div>
`;
